src/components/rangeUsePlan/RangeUsePlan.js

```diff
@@ -49,26 +52,44 @@ export class RangeUsePlan extends Component {
     this.setState({ isPendingModalOpen: false });
   }
 
+  updateStatus = (statusName, closeConfirmModal) => {
+    const { rangeUsePlan, statuses, updateRupStatus } = this.props;
+    const status = statuses.find(status => status.name === statusName);
+    const requestData = {
+      agreementId: rangeUsePlan.id,
+      statusId: status.id,
+    }
+
+    updateRupStatus(requestData).then(closeConfirmModal);
+  }
+
   onYesCompletedClicked = () => {
-    const completed = this.props.statuses.find(status => status.name === COMPLETED);
-    // console.log(completed)
-    // TODO: make a network call with completed.id
-    this.closeCompletedConfirmModal();
+    this.updateStatus(COMPLETED, this.closeCompletedConfirmModal);
+  }
+
+  onYesPendingClicked = () => {
+    this.updateStatus(PENDING, this.closePendingConfirmModal);
   }
 
   render() {
     const { isCompletedModalOpen, isPendingModalOpen } = this.state;
-
+    const { rangeUsePlan, isUpdatingStatus, newStatus } = this.props;
     const statusDropdownOptions = [
       { key: 1, text: COMPLETED, value: 1, onClick: this.openCompletedConfirmModal },
       { key: 2, text: PENDING, value: 2, onClick: this.openPendingConfirmModal },
     ];
+
     const { 
       agreementId, agreementStartDate, agreementEndDate,
-      zone, rangeName, alternateBusinessName, planStartDate, planEndDate,
-    } = this.props.rangeUsePlan;
+      zone, rangeName, alternateBusinessName, 
+      planStartDate, planEndDate, status:currStatus,
+      primaryAgreementHolder: currPAHolder,
+    } = rangeUsePlan;
```

Comment:
I think it's best to put all on one line or each on it's own line but it's more just personal preference.